themes/classic/_dev/js/cart.js

```diff
@@ -168,6 +168,36 @@ $(document).ready(() => {
     });
   };
 
+  //Hide 'have a code promo' text if displayed when 'close' is
+  let isPresent = $('.promo-code-button').length,
+      nameLength = 0;
+
+  function labelCodeDisplayed() {
+    $('.promo-code-button').not('.cancel-promo').hide();
+    
+    $('.cancel-promo').on('click', () => {
+      $('.cancel-promo').hide();
```

Comment:
```suggestion
      $(this).hide();
```

should be working, and prevent a lookup in the DOM